packages/kit/src/runtime/server/index.js

```diff
@@ -246,12 +245,7 @@ export async function respond(request, options, state) {
 					}
 				}
 
-				for (const new_cookie of Array.from(new_cookies.values())) {
-					response.headers.append(
-						'set-cookie',
-						cookie.serialize(new_cookie.name, new_cookie.value, new_cookie.options)
-					);
-				}
+				add_cookies_to_headers(response.headers, Array.from(new_cookies.values()));
```

Comment:
To clarify my previous comment, this is the buggy code, and TypeScript is ok with it because extraneous properties are OK, and all required properties are the same between these types. The difference is that this shape has the options in the property `options` while in the other cases it's part of the same object that also has name/value.